### PR TITLE
[DOCS] Rework semantic search main page

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR reworks and updates the [semantic search main page](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-search.html).
* Adds a new `Choose a semantic search workflow` section that describes the various ways users can use the Stack for semantic search and recommends the semantic text workflow
* Adds a new `Using the right query` section that helps users understand which query should be used for the different vectors
* Provides a table about the field types and the corresponding queries
* Moves the deployed ELSER and dense vector model tutorial to a sub-page (content is unchanged)
* Reorganize the sub-pages in TOC so the most important comes first
* Improves read more list at the end of the main page

### Preview

* [Semantic search](https://elasticsearch_bk_112452.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search.html)
* [Tutorial: semantic search with deployed models](https://elasticsearch_bk_112452.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-deployed-nlp-model.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 1]
```diff
@@ -20,95 +20,57 @@ image::images/search/vector-search-oversimplification.png[A simplified represent
...
 
-Once you have created the mappings for the index, you can generate text embeddings from your input text.
-This can be done by using an
-<<ingest,ingest pipeline>> with an <<inference-processor,inference processor>>.
-The ingest pipeline processes the input data and indexes it into the destination index.
-At index time, the inference ingest processor uses the trained model to infer against the data ingested through the pipeline.
-After you created the ingest pipeline with the inference processor, you can ingest your data through it to generate the model output.
+Crafting the right query is crucial for semantic search.
+The query type you should use depends first on whether you are using the recommended `semantic_text` workflow.
+If not, it depends on the vector type in which your embeddings are stored.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:46]:
```suggestion
Which query you use and which field you target in your queries depends on your chosen workflow.
If you're using the `semantic_text` workflow it's quite simple.
If not, it depends on which type of embeddings you're working with.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 2]
```diff
@@ -20,95 +20,57 @@ image::images/search/vector-search-oversimplification.png[A simplified represent
...
 
-include::{es-ref-dir}/tab-widgets/semantic-search/generate-embeddings-widget.asciidoc[]
+[cols="3*", options="header"]
+|=======================================================================================================================================================================================================
+| Field to query                    | Query to use                                      | Notes
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:50]:
```suggestion
| Field type to query                    | Query to use                                      | Notes                                                                                                                                                             
```

File: docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc [hunk 3]
```diff
@@ -0,0 +1,92 @@
+[[semantic-search-deployed-nlp-model]]
+=== Tutorial: semantic search with a deployed model
+
+++++
+<titleabbrev>Semantic search with deployed model</titleabbrev>
+++++
+
+IMPORTANT: For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.
+
+This guide shows you how to implement semantic search with models deployed in {es}: from selecting an NLP model, to writing queries.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc:10]: Maybe could have a couple of background sentences that explain that this tutorial was written before the inference endpoints and semantic_text options were introduced and therefore are a little dated. Just one or two extra lines of context might be helpful to re-route the reader.

Maybe that could all live in the IMPORTANT block too?

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 4]
```diff
@@ -20,95 +20,57 @@ image::images/search/vector-search-oversimplification.png[A simplified represent
...
+| <<sparse-vector,`sparse_vector`>> | <<query-dsl-sparse-vector-query,`sparse vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+| <<dense-vector,`dense_vector`>>   | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+|=======================================================================================================================================================================================================
 
-Now it is time to perform semantic search!
+If you want {es} to generate embeddings for you both index and query time, use the `semantic_text` field and the `semantic` query.
+If you want to bring your own embeddings, store them in {es} and use the `sparse_vector` or `dense_vector` field type and the associated query depending on the NLP model you used for generating the embeddings.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:57]: Would like to see mention of the BYOE use case earlier in this doc somewhere too.

**Review by leemthompo — COMMENTED** [review]
This is looking good and is already super helpful for outlining the options! I have a few suggestions and ideas. :)

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 5]
```diff
@@ -20,95 +21,60 @@ image::images/search/vector-search-oversimplification.png[A simplified represent
...
+If you're using the `semantic_text` workflow it's quite simple.
+If not, it depends on which type of embeddings you're working with.
 
-[discrete]
-[[semantic-search-hybrid-search]]
-=== Beyond semantic search with hybrid search
+[cols="3*", options="header"]
+|=======================================================================================================================================================================================================
+| Field type to query                    | Query to use                                      | Notes                                                                                                                                                             
+| <<semantic-text,`semantic_text`>> | <<query-dsl-semantic-query,`semantic`>> .         | The `semantic_text` field handles generating embeddings for you at index time and query time.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:55]:
```suggestion
| <<semantic-text,`semantic_text`>> | <<query-dsl-semantic-query,`semantic`>>          | The `semantic_text` field handles generating embeddings for you at index time and query time.                                                                     
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 6]
```diff
@@ -20,95 +20,57 @@ image::images/search/vector-search-oversimplification.png[A simplified represent
...
-Not only that, but hybrid search using reciprocal rank fusion {blog-ref}improving-information-retrieval-elastic-stack-hybrid[has been shown to perform better in general].
+IMPORTANT: For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.
 
-include::{es-ref-dir}/tab-widgets/semantic-search/hybrid-search-widget.asciidoc[]
 
 [discrete]
 [[semantic-search-read-more]]
 === Read more
 
 * Tutorials:
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/reference/search/search-your-data/semantic-search.asciidoc:70]: Great idea. However, I'd keep this section at the end of the page to have all the related materials in one place for ease of reference.

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 7]
```diff
@@ -9,9 +9,9 @@ Embeddings are vectors that provide a numeric representation of a text.
 Pieces of content with similar meaning have similar representations.
 NLP models can be used in the {stack} various ways, you can:
 
-* deploy models in {es}
 * use the <<semantic-search-semantic-text, `semantic_text` workflow>> (recommended)
 * use the <<semantic-search-inference, {infer} API workflow>>
+* deploy models directly in {es}
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply docs/reference/search/search-your-data/semantic-search.asciidoc:15]: I'd highlight the recommended option. If we label all three they seem like semantic search is for newcomers while the other two are for more advanced users but AFAIK it's not our intention. WDYT?

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 8]
```diff
@@ -7,108 +7,77 @@ Semantic search is a search method that helps you find data based on the intent
...
+| <<semantic-text,`semantic_text`>> .    | <<query-dsl-semantic-query,`semantic`>> .         | The `semantic_text` field handles generating embeddings for you at index time and query time.                                                                     
+| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+| <<dense-vector,`dense_vector`>>        | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+|=======================================================================================================================================================================================================
```
Comment (szabosteve) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:61]:
```suggestion
| Field type to query                    | Query to use                                      | Notes                                                                                                                                                             
| <<semantic-text,`semantic_text`>>      | <<query-dsl-semantic-query,`semantic`>>           | The `semantic_text` field handles generating embeddings for you at index time and query time.                                                                     
| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
| <<dense-vector,`dense_vector`>>        | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
|=======================================================================================================================================================================================================
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 9]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
+Which query you use and which field you target in your queries depends on your chosen workflow.
+If you're using the `semantic_text` workflow it's quite simple.
+If not, it depends on which type of embeddings you're working with.
 
-include::{es-ref-dir}/tab-widgets/semantic-search/search-widget.asciidoc[]
-
-[discrete]
-[[semantic-search-hybrid-search]]
-=== Beyond semantic search with hybrid search
+[cols="3*", options="header"]
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:64]: nit: The table columns are a bit disorienting, wonder if could give a bit more width to the first two?

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 10]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
+|=======================================================================================================================================================================================================
+| Field type to query                    | Query to use                                      | Notes                                                                                                                                                             
+| <<semantic-text,`semantic_text`>>      | <<query-dsl-semantic-query,`semantic`>>           | The `semantic_text` field handles generating embeddings for you at index time and query time.                                                                     
+| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:68]:
```suggestion
| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse_vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
```

the underscore might help with column lengths too?

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 11]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
-After you decide which model you want to use for implementing semantic search, you need to deploy the model in {es}.
+==== The `semantic_text` workflow
 
-include::{es-ref-dir}/tab-widgets/semantic-search/deploy-nlp-model-widget.asciidoc[]
+The simplest way to use NLP models in the {stack} is through the <<semantic-search-semantic-text, `semantic_text` workflow>>.
+We recommend using this approach because it abstracts away a lot of manual work.
+All you need to do is create an {infer} endpoint and an index mapping to start ingesting, embedding, and querying data.
+There is no need to define model-related settings and parameters, or to create {infer} ingest pipelines.
+Refer to the <<put-inference-api, Create an {infer} endpoint API>> documentation for a list of supported services.
+The <<semantic-search-semantic-text, Semantic search with `semantic_text`>> tutorial shows you the process end-to-end.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:37]:
```suggestion

The <<semantic-search-semantic-text, Semantic search with `semantic_text`>> tutorial shows you the process end-to-end.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 12]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
-=== Map a field for the text embeddings
+==== The {infer} API workflow
 
-Before you start using the deployed model to generate embeddings based on your input text, you need to prepare your index mapping first.
-The mapping of the index depends on the type of model.
-
-include::{es-ref-dir}/tab-widgets/semantic-search/field-mappings-widget.asciidoc[]
+The <<semantic-search-inference, {infer} API workflow>> is more complex but offers greater control over the {infer} endpoint configuration.
+You need to create an {infer} endpoint, provide various model-related settings and parameters, define an index mapping, and set up an {infer} ingest pipeline with the appropriate settings.
+The <<semantic-search-inference, Semantic search with the {infer} API>> tutorial shows you the process end-to-end.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:44]:
```suggestion

The <<semantic-search-inference, Semantic search with the {infer} API>> tutorial shows you the process end-to-end.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 13]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
-The ingest pipeline processes the input data and indexes it into the destination index.
-At index time, the inference ingest processor uses the trained model to infer against the data ingested through the pipeline.
-After you created the ingest pipeline with the inference processor, you can ingest your data through it to generate the model output.
+==== Model deployment workflow
 
-include::{es-ref-dir}/tab-widgets/semantic-search/generate-embeddings-widget.asciidoc[]
+You can also deploy NLP in {es} manually, without using an {infer} endpoint.
+This is the most complex and labor intensive workflow for performing semantic search in the {stack}.
+You need to select an NLP model from the {ml-docs}/ml-nlp-model-ref.html#ml-nlp-model-ref-text-embedding[list of supported dense and sparse vector models], deploy it using the Eland client, create an index mapping, and set up a suitable ingest pipeline to start ingesting and querying data.
+the <<semantic-search-deployed-nlp-model, Semantic search with a model deployed in {es}>> tutorial shows you the process end-to-end.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:52]:
```suggestion

The <<semantic-search-deployed-nlp-model, Semantic search with a model deployed in {es}>> tutorial shows you the process end-to-end.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 14]
```diff
@@ -7,108 +7,86 @@ Semantic search is a search method that helps you find data based on the intent
...
+| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+| <<dense-vector,`dense_vector`>>        | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. You are expected to provide embeddings at index time.
+|=======================================================================================================================================================================================================
 
-In some situations, lexical search may perform better than semantic search.
-For example, when searching for single words or IDs, like product numbers.
+If you want {es} to generate embeddings for you both index and query time, use the `semantic_text` field and the `semantic` query.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:72]:
```suggestion
If you want {es} to generate embeddings at both index and query time, use the `semantic_text` field and the `semantic` query.
```

File: docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+=== Tutorial: semantic search with a deployed model
+
+++++
+<titleabbrev>Semantic search with deployed model</titleabbrev>
+++++
+
+[IMPORTANT]
+====
+* For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.
+* This tutorial was written before the {infer} endpoint and `semantic_text` options were introduced which makes this as the most complicated option. We recommend to use another method to perform semantic search.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc:11]:
```suggestion
* This tutorial was written before the {infer} endpoint and `semantic_text` options were introduced. Today we have simpler options for performing semantic search. 
```

**Review by leemthompo — APPROVED** [review]
This is going to be very helpful! 🎉 

Few final little suggestions, and a question about whether we can make the table columns nicer.

File: docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+=== Tutorial: semantic search with a deployed model
+
+++++
+<titleabbrev>Semantic search with deployed model</titleabbrev>
+++++
+
+[IMPORTANT]
+====
+* For the easiest way to perform semantic search in the {stack}, refer to the <<semantic-search-semantic-text, `semantic_text`>> end-to-end tutorial.
+* This tutorial was written before the {infer} endpoint and `semantic_text` options were introduced. Today we have simpler options for performing semantic search.
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search-deploy-model.asciidoc:11]:
```suggestion
* This tutorial was written before the {infer} endpoint and `semantic_text` field type was introduced. Today we have simpler options for performing semantic search. 
```

Also can we add links to the [inference API](https://www.elastic.co/guide/en/elasticsearch/reference/current/inference-apis.html) and [semantic_text field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-text.html) documentation here?

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 17]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
+If you're using the `semantic_text` workflow it's quite simple.
+If not, it depends on which type of embeddings you're working with.
 
-include::{es-ref-dir}/tab-widgets/semantic-search/search-widget.asciidoc[]
+[cols="30%, 30%, 40%", options="header"]
+|=======================================================================================================================================================================================================
+| Field type to query                    | Query to use                                      | Notes                                                                                                                                                             
+| <<semantic-text,`semantic_text`>>      | <<query-dsl-semantic-query,`semantic`>>           | The `semantic_text` field handles generating embeddings for you at index time and query time.                                                               
+| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse_vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. Provide embeddings at index time.
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:71]:
```suggestion
| <<sparse-vector,`sparse_vector`>>      | <<query-dsl-sparse-vector-query,`sparse_vector`>> | The `sparse_vector` query can generate query embeddings for you, but you can also provide your own. You must provide embeddings at index time.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 18]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
+| <<dense-vector,`dense_vector`>>        | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. Provide embeddings at index time.
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:72]:
```suggestion
| <<dense-vector,`dense_vector`>>        | <<query-dsl-knn-query,`knn`>>                     | The `knn` query can generate query embeddings for you, but you can also provide your own. You must provide embeddings at index time.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 19]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
+|=======================================================================================================================================================================================================
 
-[discrete]
-[[semantic-search-hybrid-search]]
-=== Beyond semantic search with hybrid search
+If you want {es} to generate embeddings at both index and query time, use the `semantic_text` field and the `semantic` query.
+If you want to bring your own embeddings, store them in {es} and use the `sparse_vector` or `dense_vector` field type and the associated query depending on the NLP model you used for generating the embeddings.
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:76]:
```suggestion
If you want to bring your own embeddings, use the `sparse_vector` or `dense_vector` field type and the associated query depending on the NLP model you used to generate the embeddings.
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 20]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
 
 [discrete]
 [[semantic-search-read-more]]
 === Read more
 
 * Tutorials:
-** <<semantic-search-elser,Semantic search with ELSER>>
+** <<semantic-search-semantic-text, Semantic search with `semantic_text`>>
+** <<semantic-search-inference, Semantic search with the {infer} API>>
+** <<semantic-search-elser,Semantic search with ELSER>> using the {infer} workflow
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:88]:
```suggestion
** <<semantic-search-elser,Semantic search with ELSER>> using the model deployment workflow
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 21]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
-Each non-zero element of the vector corresponds to a term in the model vocabulary.
-The ELSER vocabulary contains around 30000 terms, so the sparse vectors created by ELSER contain about 30000 values, the majority of which are zero.
-Effectively the ELSER model is replacing the terms in the original query with other terms that have been learnt to exist in the documents that best match the original search terms in a training dataset, and weights to control how important each is.
+[[using-nlp-models]]
+=== Choose a semantic search workflow
 
 [discrete]
-[[semantic-search-deploy-nlp-model]]
-=== Deploy the model
+==== The `semantic_text` workflow
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:30]:
```suggestion
==== `semantic_text` workflow
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 22]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
+There is no need to define model-related settings and parameters, or to create {infer} ingest pipelines.
+Refer to the <<put-inference-api, Create an {infer} endpoint API>> documentation for a list of supported services.
 
-include::{es-ref-dir}/tab-widgets/semantic-search/deploy-nlp-model-widget.asciidoc[]
+The <<semantic-search-semantic-text, Semantic search with `semantic_text`>> tutorial shows you the process end-to-end.
 
 [discrete]
-[[semantic-search-field-mappings]]
-=== Map a field for the text embeddings
+==== The {infer} API workflow
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:41]:
```suggestion
==== {infer} API workflow
```

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 23]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
 
 * Tutorials:
-** <<semantic-search-elser,Semantic search with ELSER>>
+** <<semantic-search-semantic-text, Semantic search with `semantic_text`>>
+** <<semantic-search-inference, Semantic search with the {infer} API>>
+** <<semantic-search-elser,Semantic search with ELSER>> using the {infer} workflow
+** <<semantic-search-deployed-nlp-model, Semantic search with a model deployed in {es}>>
 ** {ml-docs}/ml-nlp-text-emb-vector-search-example.html[Semantic search with the msmarco-MiniLM-L-12-v3 sentence-transformer model]
+* Interactive examples:
+** The https://github.com/elastic/elasticsearch-labs[`elasticsearch-labs`] repo contains a number of interactive semantic search examples in the form of executable Python notebooks, using the {es} Python client
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:92]: Can we link to some specific examples here?

Semantic search with ELSER using the model deployment workflow: https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/03-ELSER.ipynb

Semantic search with `semantic_text`: https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/search/09-semantic-text.ipynb

File: docs/reference/search/search-your-data/semantic-search.asciidoc [hunk 24]
```diff
@@ -7,108 +7,89 @@ Semantic search is a search method that helps you find data based on the intent
...
 * Blogs:
```
Comment (Mikep86) [inline docs/reference/search/search-your-data/semantic-search.asciidoc:93]: Can we link to the [`semantic_text` blog post](https://www.elastic.co/search-labs/blog/semantic-search-simplified-semantic-text) in this section?

**Review by Mikep86 — COMMENTED** [review]
Great work 🙌 ! I left some minor comments about wording and some more links.

**Review by Mikep86 — APPROVED** [review]
LGTM, thank you for your patience!

**Pull request merged**